compiler/tests/compile_errors.rs

```diff
@@ -36,8 +36,6 @@ use rstest::rstest;
     case("duplicate_var_in_child_scope.fe", AlreadyDefined),
     case("duplicate_var_in_contract_method.fe", AlreadyDefined),
     case("emit_undefined_event.fe", MissingEventDefinition),
-    case("emit_without_call.fe", EventInvocationExpected),
-    case("emit_with_invalid_call.fe", EventInvocationExpected),
```

Comment:
@sbillig These two analyzer errors aren't needed anymore because the AST will simply not allow `emit` to be followed by arbitrary expressions (which we previously rejected in the analyzer pass). But do we have a way yet to check for expected parser errors so that I can ensure we won't regress in the future.